server/src/main/java/org/elasticsearch/cluster/service/MasterService.java

```diff
@@ -899,7 +899,7 @@ private static ClusterState innerExecuteTasks(
         final var taskContexts = castTaskContexts(executionResults);
         try {
             return executor.execute(previousClusterState, taskContexts);
-        } catch (Exception e) {
+        } catch (Throwable e) {
```

Comment:
Catching `Throwable` is not acceptable, sorry, you'll need to revert this.

Reply:
It's ok for a test to just hang if something throws an `Error`. Certainly not ideal, but definitely better than catching `Throwable`. There are timeouts which would terminate the test run eventually.